### PR TITLE
feat: add more enrichment plugin example

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

## Description

https://pr-887.d19s7xzcva2mw3.amplifyapp.com/data/sdk-plugins/#enrichment-type-plugin_1

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sdk-plugins.md [hunk 1]
```diff
@@ -103,18 +103,68 @@ Use an Enrichment Plugin to modify event properties:
...
+              async setup(_, amplitude) {
+                if (shouldSetUserProperties) {
+                  const identifyEvent = new amplitude.Identify();
+                  identifyEvent.set("testKey", "testValue");
+                  amplitude.identify(identifyEvent);
+                }
+                return undefined;
+              }
+            }
+
```
Comment (Mercy811) [inline docs/data/sdk-plugins.md:140]: Do we need `execute()`?

**Review by Mercy811 — COMMENTED** [review]
Curious when who is asking for this plugin and what scenario customer would use it?

File: docs/data/sdk-plugins.md (hunk 1 above)
Comment (yuhao900914) [reply docs/data/sdk-plugins.md:140]: execute is optional I think. https://www.docs.developers.amplitude.com/data/sdks/browser-2/#pluginexecute-optional-for-type-enrichment

**Comment by yuhao900914** [comment]
> Curious when who is asking for this plugin and what scenario customer would use it?

The customer want to set a user priority before any event, for example the session start event. 
Please check [this](https://discourse.amplitude.com/t/amplitude-start-session-race-conditions/7142/14?u=alyssa_yu) discourse question.

**Review by Mercy811 — APPROVED** [review]

File: docs/data/sdk-plugins.md (hunk 1 above)
Comment (yuhao900914) [reply docs/data/sdk-plugins.md:140]: But I notice that execute is not optional in the order version of the SDK. Thanks for calling that out.  I will add that back.

**Review by bgiori — APPROVED** [review]

**Pull request merged**